README.md

```diff
@@ -121,20 +148,79 @@ area of the system the error is emerging from.
 **Wrap**: `Module(module string)`: It returns a default `ModuleError` implementation
 which is also a `ErrorResponseAggregator`.
 
-**ErrorWithCode**
+**Usage**:
 
-This error means something special, it marks the error can be reported to the
-final client. Otherwise, the idea is that an unknown error blows to the actor
-which is consuming the system.
+```go
+// using predefined module
+var (
+    ModuleCatalog = errors.Module("catalog")
+)
 
-**Wrap**: `Code(module string)`: It returns a default `CodeError` implementation
+errors.Wrap(err, ModuleCatalog)
+```
+
+#### ErrorWithCode
+
+This error adds the `Code` information to the error. The code refers to a (documented)
+known error.
+
+**Wrap**: `Code(code string)`: It returns a default `CodeError` implementation
 which is also a `ErrorResponseAggregator`.
 
-**ValidationError**
+**Usage**:
+
+```go
+// using predefined code
+var (
+    CodeProductNotFound = errors.Code("product-not-found")
+)
 
-A special case error for dealing with validation errors from the 
+errors.Wrap(err, CodeProductNotFound)
+```
+
+#### ValidationError
+
+A special case error for dealing with validation errors from the
```

Comment:
It would be nice to add the repository link.

Reply:
Done!